src/ceph-volume/ceph_volume/devices/lvm/strategies/bluestore.py

```diff
@@ -15,18 +15,23 @@ class SingleType(Strategy):
     Support for all SSDs, or all HDDS
     """
 
-    def __init__(self, devices, args):
-        super(SingleType, self).__init__(devices, args)
+    def __init__(self, data_devs, db_devs, wal_devs, args):
+        super(SingleType, self).__init__(data_devs, db_devs, wal_devs, args)
         self.validate_compute()
 
+    @classmethod
+    def with_auto_devices(cls, devices, args):
```

Comment:
this seems to have been added for test purposes only, was it not possible to declare the auto/non-auto behavior via an argument? Maybe more difficult now that the args moved to __init__

Reply:
This method is used here: https://github.com/ceph/ceph/blob/c29d9c59d5eab40f9dc3771e5e20e3e3dac82f29/src/ceph-volume/ceph_volume/devices/lvm/batch.py#L290

> was it not possible to declare the auto/non-auto behavior via an argument? Maybe more difficult now that the args moved to init

Not sure what you mean here? Using auto or no-auto is of course decided by the arguments. When auto is desired the drives still have to be split at some point.

Reply:
I think the question is why do we need this class method when the class should be able to figure out if it needs the auto behavior or not by inspecting ``self.args``?

Reply:
In the existing code `batch` inspects args and figures out the strategy, I just kept with that. That could certainly be implemented differently, though I'd argue to find consensus on this feature and refactor things in a later PR.